datadog_checks_base/datadog_checks/checks/openmetrics/mixins.py

```diff
@@ -0,0 +1,747 @@
+        :param pFormat: the preferred format defined in PrometheusFormat
+        :param headers: extra headers
+        :return: requests.Response
+        """
+        endpoint = scraper_config.get('prometheus_url')
+
+        # Should we send a service check for when we make a request
+        health_service_check = scraper_config['health_service_check']
+        service_check_name = '{}{}'.format(scraper_config['NAMESPACE'], '.prometheus.health')
+        service_check_tags = scraper_config['custom_tags'] + ['endpoint:' + endpoint]
+        try:
+            response = self.send_request(endpoint, scraper_config, pFormat, headers)
+        except requests.exceptions.SSLError:
+            self.log.error("Invalid SSL settings for requesting {} endpoint".format(endpoint))
+            raise
+        except IOError:
+            if health_service_check:
+                self.service_check(
+                    service_check_name,
+                    AgentCheck.CRITICAL,
+                    tags=service_check_tags
+                )
+            raise
+        try:
+            response.raise_for_status()
+            if health_service_check:
+                self.service_check(
+                    service_check_name,
+                    AgentCheck.OK,
+                    tags=service_check_tags
+                )
+            return response
+        except requests.HTTPError:
+            response.close()
+            if health_service_check:
+                self.service_check(
+                    service_check_name,
+                    AgentCheck.CRITICAL,
+                    tags=service_check_tags
+                )
+            raise
+
+    def send_request(self, endpoint, scraper_config, pFormat=PrometheusFormat.PROTOBUF, headers=None):
+        # Determine the headers
+        if headers is None:
+            headers = {}
+        if 'accept-encoding' not in headers:
+            headers['accept-encoding'] = 'gzip'
+        if pFormat == PrometheusFormat.PROTOBUF:
+            headers['accept'] = 'application/vnd.google.protobuf; ' \
+                                'proto=io.prometheus.client.MetricFamily; ' \
+                                'encoding=delimited'
+        headers.update(scraper_config['extra_headers'])
+
+        # Determine the SSL verification settings
+        cert = None
+        if isinstance(scraper_config['ssl_cert'], string_types):
+            if isinstance(scraper_config['ssl_private_key'], string_types):
+                cert = (scraper_config['ssl_cert'], scraper_config['ssl_private_key'])
+            else:
+                cert = scraper_config['ssl_cert']
+        verify = True
+        if isinstance(scraper_config['ssl_ca_cert'], string_types):
+            verify = scraper_config['ssl_ca_cert']
+        elif scraper_config['ssl_ca_cert'] is False:
+            disable_warnings(InsecureRequestWarning)
+            verify = False
+
+        # Determine the authentication settings
```

Comment:
Yes, this is the only new thing I've added to the base class. It is not currently used by a check, but I think this might be useful for future checks or feature requests.

Reply:
I guess one could use the HTTP extra headers for this as well?